src/bindgen/ir/ty.rs

```diff
@@ -519,59 +519,121 @@ impl Type {
+                        let mut did_erase_generics = false;
+                        let generics: Vec<_> = path
+                            .generics()
+                            .iter()
+                            .map(|g| match g {
+                                GenericArgument::Type(ty) => {
+                                    let erased_ty =
+                                        eraser.erase_transparent_types(library, ty, mappings);
+                                    if erased_ty.is_some() {
+                                        did_erase_generics = true;
+                                    };
+                                    GenericArgument::Type(erased_ty.unwrap_or_else(|| ty.clone()))
+                                }
+                                other => other.clone(),
+                            })
+                            .collect();
+
+                        // Replace transparent items with their underlying type and erase it.
+                        let aliased_ty = match item {
+                            ItemContainer::OpaqueItem(o) => o.as_transparent_alias(&generics),
+                            ItemContainer::Typedef(t) => t.as_transparent_alias(&generics),
+                            ItemContainer::Struct(s) => s.as_transparent_alias(&generics),
+                            _ => None,
+                        };
+                        if let Some(mut ty) = aliased_ty {
+                            eraser.erase_transparent_types_inplace(library, &mut ty, mappings);
+                            return Some(ty);
+                        } else if did_erase_generics {
+                            // The type was not transparent, but some of its generics were erased.
+                            let path = GenericPath::new(path.path().clone(), generics);
+                            return Some(Type::Path(path));
+                        }
+                    }
+                } else {
+                    warn!(
+                        "Can't find {}. This usually means that this type was incompatible or \
+                           not found.",
+                        path.path()
+                    );
+                }
+            }
+            Type::Primitive(..) => {} // cannot simplify further
+            Type::Array(ref ty, ref constexpr) => {
+                if let Some(erased_ty) = eraser.erase_transparent_types(library, ty, mappings) {
+                    return Some(Type::Array(Box::new(erased_ty), constexpr.clone()));
+                }
+            }
+            Type::FuncPtr {
+                ref ret,
+                ref args,
+                is_nullable,
+                never_return,
+            } => {
+                // Predict that ret+args will all be erased, but decrement the count whenever we're
+                // wrong. If the count drops to 0, then type erasure was a no-op after all.
```

Comment:
Could you elaborate on what this is trying to handle? An example would be great.

Reply:
Disregard this, this was about me not groking that the logic was basically ("try and find if we erased any")